habitat_baselines/il/trainers/pacman_trainer.py

```diff
@@ -378,208 +381,250 @@ def _eval_checkpoint(
+                                i,
+                                config.IL.NAV.max_controller_actions,
+                            )
+                            if j == "pred":
+                                planner_actions_in = planner_actions_in.to(
+                                    self.device
+                                )
+                                planner_img_feats = planner_img_feats.to(
+                                    self.device
+                                )
 
-                for i in [10, 30, 50]:
-                    if i > action_length.item():
-                        invalids.append([idx.item(), i])
-                        continue
-
-                    question = question.to(self.device)
-
-                    controller_step = False
-                    planner_hidden = model.planner_nav_rnn.init_hidden(1)
-                    # get hierarchical action history
-                    (
-                        planner_actions_in,
-                        planner_img_feats,
-                        controller_step,
-                        controller_action_in,
-                        controller_img_feats,
-                        init_pos,
-                        controller_action_counter,
-                    ) = nav_dataset.get_hierarchical_features_till_spawn(
-                        idx.item(),
-                        actions[0, : action_length[0]].numpy(),
-                        i,
-                        config.IL.NAV.max_controller_actions,
-                    )
-
-                    planner_actions_in = planner_actions_in.to(self.device)
-                    planner_img_feats = planner_img_feats.to(self.device)
-
-                    # forward planner till spawn to update hidden state
-                    for step in range(planner_actions_in.size(0)):
-
-                        planner_scores, planner_hidden = model.planner_step(
-                            question,
-                            planner_img_feats[step].unsqueeze(0).unsqueeze(0),
-                            planner_actions_in[step].view(1, 1),
-                            planner_hidden,
+                                for step in range(planner_actions_in.size(0)):
+
+                                    (
+                                        planner_scores,
+                                        planner_hidden,
+                                    ) = model.planner_step(
+                                        question,
+                                        planner_img_feats[step][
+                                            (None,) * 2
+                                        ],  # unsqueezing twice
+                                        planner_actions_in[step].view(1, 1),
+                                        planner_hidden,
+                                    )
+                        else:
+                            random_pos = env.sim.sample_navigable_point()
```

Comment:
Keeping this deterministic will be very hard and this random point is not guaranteed to be part of the same connected component of any instance of the object in question.  If you just `get_hierarchical_features_till_spawn` 0 steps, that should work for random init.  The initial agent heading and position is random.

Reply:
Thanks @erikwijmans. I've changed this in the latest commit.